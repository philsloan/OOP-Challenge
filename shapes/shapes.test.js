const { Triangle, Circle, Square } = require('./shapes');

// Mocking console.log to suppress output during tests
jest.spyOn(console, 'log').mockImplementation(() => {});

// Shape Test Suite
describe('Shape Classes', () => {
  afterEach(() => {
    console.log("Restore after each"); // Restore the original console.log after each test
  });
// Takes a snapshot and saves a correctly rendering app
  const testShapeRendering = (shape, expectedOutput) => {
    test('renders correctly', () => {
      expect(shape.render()).toMatchSnapshot();
    });
  };

  // Triangle Shape
  describe('Triangle', () => {
    testShapeRendering(new Triangle('ABC', 'red', 'white'));
  });

  // Circle Shape
  describe('Circle', () => {
    testShapeRendering(new Circle('123', 'blue', 'black'));
  });

  // Square Shape
  describe('Square', () => {
    testShapeRendering(new Square('XYZ', 'green', 'yellow'));
  });
});
