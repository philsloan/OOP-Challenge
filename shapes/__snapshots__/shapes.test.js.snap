// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shape Classes Circle renders correctly 1`] = `
"<svg version="1.1" width="300" height="200" xmlns="http://www.w3.org/2000/svg">
      <circle cx="150" cy="100" r="80" fill="blue" />
      <text x="150" y="125" font-size="60" text-anchor="middle" fill="black">123</text>
    </svg>"
`;

exports[`Shape Classes Square renders correctly 1`] = `
"<svg version="1.1" width="300" height="200" xmlns="http://www.w3.org/2000/svg">
      <rect width="200" height="200" fill="green"/>
      <text x="100" y="125" font-size="70" text-anchor="middle" fill="yellow">XYZ</text>
    </svg>"
`;

exports[`Shape Classes Triangle renders correctly 1`] = `
"<svg version="1.1" width="300" height="200" xmlns="http://www.w3.org/2000/svg">
      <polygon points="150, 18 244, 182 56, 182" fill="red" />
      <text x="150" y="145" font-size="50" text-anchor="middle" fill="white">ABC</text>
    </svg>"
`;
